### PR TITLE
refactor: refactor helm training session

### PR DESCRIPTION
**Pull request by ktoublanc**

### CODE REVIEW

File: day-1/ex-4/README.md [hunk 1]
```diff
@@ -1,49 +1,81 @@
...
-You can see that there are very few differences.
 
-The steps are:
+## Bonus: play with functions
 
-1. Create a second values file to overload what is different from default values
-1. Customize your chart to add environment variables
+> Objectives: Create a template to add environment variables to you chart
 
+1. Create a function to add environment variables to the deployment in the `_helpers.tpl` file.
```
Comment (yesteph) [inline day-1/ex-4/README.md]:
```suggestion
1. Create a named template to add environment variables to the deployment in the `_helpers.tpl` file.
```

File: day-1/ex-4/README.md [hunk 2]
```diff
@@ -1,49 +1,81 @@
...
+
+```sh
+helm uninstall <REPLACE BY MY APP NAME>
+```
 
-Now we want to install a second instance of your chart to deploy the equivalent of frontend-*.yaml descriptors.
-You can see that there are very few differences.
 
-The steps are:
+## Bonus: play with functions
```
Comment (yesteph) [inline day-1/ex-4/README.md]:
```suggestion
## Bonus: play with functions and named templates
```

File: day-1/ex-4/README.md [hunk 3]
```diff
@@ -1,49 +1,81 @@
...
 
-1. Create a second values file to overload what is different from default values
-1. Customize your chart to add environment variables
+> Objectives: Create a template to add environment variables to you chart
```
Comment (yesteph) [inline day-1/ex-4/README.md]: Ajouter lien vers doc ? https://helm.sh/docs/chart_template_guide/named_templates/

File: day-1/ex-4/README.md (hunk 1 above)
Comment (ktoublanc) [reply day-1/ex-4/README.md]: Applied, thanks

File: day-1/ex-4/README.md (hunk 2 above)
Comment (ktoublanc) [reply day-1/ex-4/README.md]: Applied, thanks

File: day-1/ex-4/README.md (hunk 3 above)
Comment (ktoublanc) [reply day-1/ex-4/README.md]: Applied, thanks

**Pull request merged**